src/module/item/physical/document.ts

```diff
@@ -739,6 +739,55 @@ abstract class PhysicalItemPF2e<TParent extends ActorPF2e | null = ActorPF2e | n
     ): Promise<boolean | void> {
         if (!changed.system) return super._preUpdate(changed, operation, user);
 
+        // If this weapon is gaining traits that change between PF2E/SF2e mechanics, warn before continuing
+        // The actual subclasses handle the deletion itself
+        const newTraits: string[] | undefined = changed.system.traits?.value;
+        if (this.isOfType("weapon", "armor") && newTraits) {
+            const beforeSF2eTraits = this._source.system.traits.value.filter((t) => t === "tech" || t === "analog");
+            const wasSF2e = !!beforeSF2eTraits.length;
+            const becomingSF2e = newTraits.some((t) => t === "tech" || t === "analog");
+            const runes = this._source.system.runes;
+            const hasRunes =
+                ("potency" in runes && runes.potency) ||
+                ("striking" in runes && runes.striking) ||
+                ("resilient" in runes && runes.resilient) ||
+                runes.property.length > 0;
+            const hasGrade = this.system.grade && this.system.grade !== "commercial";
+            if (wasSF2e !== becomingSF2e && (hasRunes || hasGrade)) {
+                const key = `PF2E.Item.Physical.ChangeEquipmentSystem.${becomingSF2e ? "ToStarfinder" : "ToPathfinder"}`;
+                const removedTrait = beforeSF2eTraits.find((t) => !newTraits.includes(t));
+                const otherTrait = removedTrait === "tech" ? "analog" : "tech";
+                const newTrait = newTraits.find((t) => t === "tech" || t === "analog");
```

Comment:
```suggestion
                const newTrait = newTraits.find((t) => ["tech", "analog"].includes(t));
```
Or was there a reason for expanding this out?

Reply:
This one is outdated, but the reason was stricter typing (and marginally better perf). I can swap it over since that doesn't matter too much.